### PR TITLE
npctalk to JSON: Refactor npctalk.cpp to allow more flexible effects and trials

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
```SUMMARY: Infrastructure "Refactor npctalk.cpp to allow more flexible effects and trials."```

#### Purpose of change
In preparation for moving much of npc dialogue to JSON, expand response effects based somewhat on the structure of dynamic lines. Make all currently existing talk_functions available as responses, and add some generic functions to add traits, effects, and items to NPCs and the player. Add the ability to chain effects in a simple array:

```
{ "topic": "TALK_EVAC_GUARD3_HOSTILE", "effect": [ { "u_faction_rep": -15 }, { "npc_change_faction": "hells_raiders" } ] }
```

Also add a way to specify modifiers based on the NPC's personality traits and opinion of the player to the difficulty of trial, like so:

```
"trial": { "type": "PERSUADE", "difficulty": 0, "mod": [ [ "TRUST", 3 ], [ "VALUE", 3 ], [ "ANGER",
 -3 ] ] }
```

#### Describe the solution
Add a lot of additional effects. Add parsing for arrays of effects. Add the optional `mod` parameter to trials and add parsing of simple arrays of modifiers.

It's based on the existing effects lambda infrastructure, expanded on the same model as dynamic_lines.

#### Alternative Solutions
Several npc talk_functions can be obsoleted with the new, more flexible structure, but I'll clean them up after moving everything into JSON.

#### Additional context
The full current implementation of the entire system, including NPC dialogue in JSON, is at #25432 but it is much too large of a commit to review sensibly.

### CODE REVIEW

**Comment by mlangsdorf** [comment]
Jenkins rebuild

**Comment by nexusmrsep** [comment]
It's absolutely fantastic. While I know this is not the end of the road here, I want to use this opportunity to ask you politely to add a documentation for all those inventions at the end, especially with some examples, like a fake/real quest that uses those neat tricks, for those of us who would like to build quest lines for the game in the future.

File: src/npctalk_funcs.cpp [hunk 1]
```diff
@@ -213,6 +214,118 @@ void talk_function::insult_combat( npc &p )
...
+                bionic_names.push_back( bio->tname() + " - $" + to_string( bio->price( true ) * 2 / 100 ) );
+            }
+        }
+    }
+    // Choose bionic if applicable
+    int bionic_index = 0;
+    bionic_names.push_back( _( "Cancel" ) );
+    bionic_index = menu_vec( false, _( "Which bionic do you wish to have installed?" ),
+                             bionic_names ) - 1;
+    if( bionic_index == ( int )bionic_names.size() - 1 ) {
```
Comment (ZhilkinSerg) [inline src/npctalk_funcs.cpp]: Please use `static_cast<int>` instead of C-style `( int )` cast.

Comment (mlangsdorf) [reply src/npctalk_funcs.cpp]: Fixed.

File: src/npctalk_funcs.cpp [hunk 2]
```diff
@@ -213,6 +214,118 @@ void talk_function::insult_combat( npc &p )
...
+                }
+            }
+        }
+    }
+    // Choose bionic if applicable
+    int bionic_index = 0;
+    bionic_names.push_back( _( "Cancel" ) );
+    bionic_index = menu_vec( false, _( "Which bionic do you wish to uninstall?" ),
+                             bionic_names ) - 1;
+    if( bionic_index == ( int )bionic_names.size() - 1 ) {
```
[the remark this answers is not in the document]
Comment (mlangsdorf) [reply src/npctalk_funcs.cpp]: Fixed.

File: src/npctalk.cpp [hunk 3]
```diff
@@ -2244,6 +2264,34 @@ void dialogue::gen_responses( const talk_topic &the_topic )
...
+    npc &p = *d.beta;
+    int modifier = 0;
+    if( attribute == "ANGER" ) {
+        modifier = p.op_of_u.anger;
+    } else if( attribute == "FEAR" ) {
+        modifier = p.op_of_u.fear;
+    } else if( attribute == "TRUST" ) {
+        modifier = p.op_of_u.trust;
+    } else if( attribute == "VALUE" ) {
+        modifier = p.op_of_u.trust;
```
Comment (geophree) [inline src/npctalk.cpp]: Reading the docs you added above, it doesn't seem like this should be the same as `"TRUST"`.

**Pull request merged**